torch/csrc/jit/script/convert_to_ssa.cpp

```diff
@@ -265,21 +246,120 @@ struct SSATransformer {
   }
 
   void run(std::shared_ptr<Graph>& graph) {
-    convertBlockToSSA(graph->block());
+    eraseBlockLoadsStores(graph->block());
   }
 
   std::shared_ptr<ValueEnvironment> environment_stack = nullptr;
 };
 
-// Converting to SSA works in multiple parts. First we inline the loop condition
-// before and into the body of loops, then we add outputs to control flow
-// nodes, then we stitch together Loads & Stores into SSA form.
+// This pass transforms Breaks & Continues to be LoopExit continuations,
+// of the form LoopExit(%loop_continue_condition, *loop_carried_vars)
```

Comment:
I have it working locally, it's a small change. You have a target block that an exit is targeting and you do the transforms until the exit values reach the target block. So far the exit is only the loop block, but with returns it will be targeting the graph block.